src/analyzer/protocol/websocket/events.bif

```diff
@@ -0,0 +1,71 @@
+## Generated when a WebSocket handshake completed.
+##
+## This is a bit artificial. It can be used to configure the WebSocket
+## analyzer if the HTTP headers contained protocol and extension headers.
+##
+## c: The WebSocket connection.
+##
+## aid: The analyzer identifier of the WebSocket analyzer.
+##
+## .. zeek:see:: WebSocket::__configure_analyzer
+##
+## .. zeek:see:: WebSocket::configure_analyzer
+##
+event websocket_established%(c: connection, aid: count%);
+
+## Generated for every WebSocket frame.
+##
+## c: The WebSocket connection.
+##
+## is_orig: True if the frame is from the originator, else false.
+##
+## fin: True if the fin bit is set, else false.
+##
+## rsv: The value of the RSV1, RSV2 and RSV3 bits.
+##
+## opcode: The frame's opcode.
+##
+## payload_len: The frame's payload length.
+##
+event websocket_frame%(c: connection, is_orig: bool, fin: bool, rsv: count, opcode: count, payload_len: count%);
+
+## Generated for every chunk of WebSocket frame payload data.
+##
+## Do not use it to extract data from a WebSocket connection unless for testing
+## or experimentation. Consider implementing a proper analyzer instead.
+##
+## c: The WebSocket connection.
+##
+## is_orig: True if the frame is from the originator, else false.
+##
+## data: One data chunk of frame payload. The length of is at most
+##       :zeek:see:`WebSocket::payload_chunk_size` bytes. A frame with
+##       a longer payload will result in multiple events events.
+##
+## .. zeek:see:: WebSocket::payload_chunk_size
+event websocket_frame_data%(c: connection, is_orig: bool, data: string%);
```

Comment:
I was wondering here whether users of this event would benefit from knowing whether the frame carrying this payload is text or binary.

Reply:
Hmm, maybe. It might also be payload of a close frame though. And, the type is just the opcode of the first frame with subsequent frames for the same message having an opcode of `0` (continuation). I'd push it to the users to handle `websocket_frame` to understand what to do with subsequent `_data` events (they may need to anyhow to understand message boundaries).

The whole motivation for that event was debugging, too :-)

Reply:
> And, the type is just the opcode of the first frame with subsequent frames for the same message having an opcode of 0.

Ah yeah, true. I agree, let's leave as is.